### ISSUE TITLE
Error: Test Folder ... contains a corrupt app.json

### ISSUE THREAD

**Issue by pzentner** [microsoft/AL-Go#234]
I have a repo with multiple apps and when running the pipeline, if fails with below error in the AnalyzeRepo stage. I have compared encoding, have compared any special characters, etc. and can't find any differences between this one and the other app,json files.

Any suggestions on where to look or what this could be? It seems like it's trying to go through the dependencies of the app.json and something seems to not work as the error seems to be only reported in the catch block of the function:

Why would it try to find the entire app.json as a property somewhere?

RunPipeline action failed. Error: Test folder .\MyApp.Test, specified in .AL-Go\settings.json, contains a corrupt app.json file. Error is The property '@***id=b0fdf3e6-dbfa-4916-90d3-f75f41c03ce3; name=MyApp.Test; publisher=NAV-X; version=1.0.0.0; brief=BlahBlahBlah (and blah/blah); description=BlahBlahBlah (and blah/blah); privacyStatement=https://nav-x.com/nav-x-privacy-policy/; EULA=https://nav-x.com/nav-x-license-agreement/; help=https://docs.nav-x.com/en/business-central/; url=http://nav-x.com; logo=./AppLogo.png; contextSensitiveHelpUrl=https://docs.nav-x.com/***0***/business-central/; dependencies=System.Object[]; screenshots=System.Object[]; platform=1.0.0.0; application=20.0.0.0; target=Cloud; idRanges=System.Object[]; resourceExposurePolicy=; runtime=9.0; supportedLocales=System.Object[]; features=System.Object[]***.Id' cannot be found on this object. Verify that the property exists.. Stacktrace: at <ScriptBlock>, C:\github-01-1\_work\_actions\microsoft\AL-Go-Actions\v2.0\AL-Go-Helper.ps1: line 652 at <ScriptBlock>, C:\github-01-1\_work\_actions\microsoft\AL-Go-Actions\v2.0\AL-Go-Helper.ps1: line 612 at <ScriptBlock>, C:\github-01-1\_work\_actions\microsoft\AL-Go-Actions\v2.0\AL-Go-Helper.ps1: line 584 at AnalyzeRepo, C:\github-01-1\_work\_actions\microsoft\AL-Go-Actions\v2.0\AL-Go-Helper.ps1: line 565 at <ScriptBlock>, C:\github-01-1\_work\_actions\microsoft\AL-Go-Actions\v2.0\RunPipeline\RunPipeline.ps1: line 77 at <ScriptBlock>, C:\github-01-1\_work\_temp\6eabdbc6-08cb-419b-8a22-023300a4dd41.ps1: line 1 at <ScriptBlock>, <No file>: line 1

**Comment by freddydk** [comment]
Looks like a bug.
If you can give me access to the repo, I will troubleshoot and fix (and try to give a better error if this is an error in the app.json or like)

**Comment by pzentner** [comment]
Thank you!

Just gave you access. In the past action runs, you should be able to see the issue.

**Comment by freddydk** [comment]
Is it OK if I temporarily change the repo to use a developer version of AL-Go to debug?
We can always change it back again

**Comment by pzentner** [comment]
Yes, no problem.

**Comment by freddydk** [comment]
Got back to this now - seems like I don't have permissions to run workflows:-(

**Comment by pzentner** [comment]
When I invited you, I gave you the admin role. It still says that your invitation is pending. 

It seems I invited you to the org and didn't wait with adding you to the repo. I gave you admin access on the repo.

**Comment by freddydk** [comment]
After using my dev version of AL-Go - it fails with another error.
That error indicates that the apps in the repo cannot co-exist (overlapping object IDs)
It looks like you would have to create a multi-project repo, which looks like this;

project1\
- .AL-Go\
- app1\
- app2\
project2\
- .AL-Go\
- app3\
- app4\

Basically a folder per project - each with a .AL-Go folder and the apps that are inter-dependant.

Hope that makes sense.

**Comment by pzentner** [comment]
Thank you. It does make sense and is definitely a better error message lol.

So, does this mean I cannot define the same ID range  in multiple apps? As they're PTEs, having the same range just made it easier, because object uniqueness is handled differently.Or does it actually check the IDs of all objects in the repo?

**Comment by freddydk** [comment]
Yes you can, but then the PTEs needs to be in different projects or different repositories.
All apps in one project will be built, published and tested together.

**Comment by pzentner** [comment]
So, I went through all apps to make sure that the app.json has different ID ranges for each of the apps. 

Also reviewed the apps and can't find any that have the same object IDs, field IDs, or ID ranges.

However, I am still getting the error.

I ran update system files again to see, if that would do anything. Nope
Completed the pull request and ran CI/CD in main only. Nope

I can't understand it, especially, since I have multiple repos that are the same way (multiple apps, multiple PTEs) and they work without a problem.
